helm-app-operator/pkg/helm/engine.go

```diff
@@ -49,24 +53,34 @@ func (o *OwnerRefEngine) Render(chart *chart.Chart, values chartutil.Values) (ma
 }
 
 // addOwnerRefs adds the configured ownerRefs to a single rendered file
+// Adds the ownerrefs to all the documents in a YAML file
 func (o *OwnerRefEngine) addOwnerRefs(fileContents string) (string, error) {
-	parsed := chartutil.FromYaml(fileContents)
-	if errors, ok := parsed["Error"]; ok {
-		return "", fmt.Errorf("error parsing rendered template to add ownerrefs: %v", errors)
-	}
+	var outBuf bytes.Buffer
+	encoder := yaml.NewEncoder(&outBuf)
+	manifests := releaseutil.SplitManifests(fileContents)
 
-	// Empty input
-	if len(parsed) == 0 {
-		return "", nil
-	}
+	for _, manifest := range manifests {
+		manifestMap := chartutil.FromYaml(manifest)
 
-	unst, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&parsed)
-	if err != nil {
-		return "", err
+		if errors, ok := manifestMap["Error"]; ok {
+			return "", fmt.Errorf("error parsing rendered template to add ownerrefs: %v", errors)
+		}
+
+		unst, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&manifestMap)
+		if err != nil {
+			return "", err
+		}
+
+		unstructured := &unstructured.Unstructured{Object: unst}
+		unstructured.SetOwnerReferences(o.refs)
+
+		err = encoder.Encode(unstructured.Object)
```

Comment:
@shettyh Can you change the encoder over to use something like:
```go
outBuf.WriteString(chartutil.ToYaml(unstructured.Object))
```
You may also have to add newlines and `---` separators back in as well.

I think it's probably better to inherit helm's implementation, which uses `github.com/ghodss/yaml` to do the yaml encoding.

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file

Reply:
@joelanford  Added changes to use `WriteString` instead of encoder and added unit test case for multi document yaml file